### ISSUE TITLE
private method should be set accessible before invoked

### ISSUE THREAD

**Issue by jearton** [mvel/mvel#252]
![image](https://user-images.githubusercontent.com/10370433/100055943-b518bc00-2e5f-11eb-82a9-99b9440fa463.png)

The declared methods contains private method.

mvel2 version: 2.4.0.Final

**Comment by MalcolmOdd** [comment]
@jearton can you expand on what exactly is the issue? There is code to allow invocation of private methods:
https://github.com/mvel/mvel/blob/97071e83fbafb5b25360756f0f551837a648a07c/src/main/java/org/mvel2/PropertyAccessor.java#L601

**Comment by MalcolmOdd** [comment]
Please note that invoking non-public methods may sometimes fail under high multi-thread load since "setAccessible(false)" is invoked immediately after invoking "setAccessible(false)" and making the call (In PropertyAccessor). The call to setAccessible(false) is probably not required since you get a new instance with its own accessibility state each time you retrieve the method.

**Comment by jearton** [comment]
> @jearton can you expand on what exactly is the issue? There is code to allow invocation of private methods:
> https://github.com/mvel/mvel/blob/97071e83fbafb5b25360756f0f551837a648a07c/src/main/java/org/mvel2/PropertyAccessor.java#L601

```java
public class Service {
      private void hello() {System.out.println("hello world");}
}


@Test
public void test() {
      Map<String, Object> vars = new HashMap<>();
      vars.put("service", new Service());

     MVEL.eval("service.hello()", vars);
}
```

**Comment by MalcolmOdd** [comment]
Thanks. I wasn't thinking about the direct mode (I generally use MVEL.compileExpression() - and these modes have completely different implementations). 
I performed the test with compiled mode and it failed with a MVEL "unable to resolve" message, so I think the direct mode should do the same (instead of  throwing the "unable to invoke method" as it does now).
The code that I referred to before does not come into play, it appears to be invoked when the method is public but the class is not.
Bottom line: To be consistent this test should fail but with a different exception.
